src/golang/lib/execution_environment/execution_environment.go

```diff
@@ -162,13 +163,23 @@ func IsCondaConnected(
 	return len(integrations) > 0, nil
 }
 
-// `SyncEnvsWithDB` keep args `envs` in-sync with DB.
+// Best-effort to delete all envs and log any error
+func deleteEnvs(envs []ExecutionEnvironment) {
+	for _, env := range envs {
+		err := env.DeleteEnv()
+		if err != nil {
+			log.Errorf("Failed to delete env %s: %v", env.Id.String(), err)
+		}
+	}
+}
+
+// CreateMissingAndSyncExistingEnvs env %s: %vistingEnvs` keep argerrsync with DB.
```

Comment:
@likawind  The comment here seems to be messed up...I wrote a proposal before maybe we can just use that one instead.